### ISSUE TITLE
[Bug] Private Theme is not applied correctly when opening from external app

### ISSUE THREAD

**Issue by Meshikhah** [mozilla-mobile/fenix#18496]
## Steps to reproduce
Enable open external link in private tab
Open a link from twitter then open it in Firefox nightly.
Open Enhanced tracking protection.

### Expected behavior
it should be readable 
### Actual behavior
It is white and unreadable
<img src=https://user-images.githubusercontent.com/29687018/111299516-42795980-8665-11eb-8d38-863f99a65082.jpg width = 45%>
### Device information

* Device vendor / model and Android version: ? j7 prime
* Firefox for Android version: ? (go to Settings -> About Firefox) 
Firefox Nightly 210315 17:01 (Build #2015798857)
AC: 74.0.20210313190118, 05dcfd7b3
GV: 88.0a1-20210313094300
AS: 73.0.1

**Comment by mcarare** [comment]
I have not been able to reproduce this. Can QA also take a look? Thank you!

**Comment by abodea** [comment]
I was not able to reproduce this issue on the latest Nightly 3/19 with Google Pixel 4 XL (11), Samsung Galaxy S10+ (10), Samsung Galaxy S7 (7).
Note that I tried with light and dark them setup but with 0 success on reproducing it.
@Meshikhah  can you please provide more information and also a video/gif?
Until further notice, I will remove `eng:qa:needed`.

**Comment by Meshikhah** [comment]
@abodea i can't record the screen because it opens in private tab.

**Comment by Amejia481** [comment]
@Meshikhah you can enable this under **Settings** -> **Private Browsing** -> **Allow screenshots in private browsing**.

**Comment by Meshikhah** [comment]
Here is the [video](http://imgur.com/gallery/GmGTJp6)

**Comment by mcarare** [comment]
@Meshikhah Thank you for the video!
I still cannot reproduce this following the steps in shown. 
What setting do you have for Theme? Does this reproduce with other settings for Theme options? Thank you.

**Comment by Meshikhah** [comment]
@mcarare I am using light theme. And i have enabled "open links in private tab"

**Comment by mcarare** [comment]
@Meshikhah What Android version does your J7 prime phone have?

**Comment by Meshikhah** [comment]
Andorid 8.1.0

**Comment by sflorean** [comment]
I was able to reproduce this issue on Nightly 3/24, following the steps/video with LG G7 Fit (Android 8.1.0).

**Comment by mcarare** [comment]
Also noticed that stats bar color is also wrong, not just the preference screen background.

**Comment by mcarare** [comment]
Eng note:  When switching from external app to browser fragment the `windowBackground` is the one from the current theme applied to the app (light or dark) and not from the private theme. This is the color used for the preference screen background and it's the color displayed on the status bar since the status bar is transparent.

**Comment by mcarare** [comment]
Reopened for QA.

**Comment by ebalazs-sv** [comment]
Verified as fixed on Nightly 4/20 with Motorola Moto G6 (Android 8).
